### ISSUE TITLE
Error message when removing value from number knob

### ISSUE THREAD

**Issue by CLL80** [storybooks/storybook#2475]
### Issue details

There is an error message when removing the value from a "number" knob input:
```
Warning: `value` prop on `input` should not be null. Consider using an empty string to clear the component or `undefined` for uncontrolled components.
```
I would expect this to be handled by the knob itself so we can avoid having these error messages in our consoles.

### Steps to reproduce

This issue can be seen in the `cra-kitchen-sink` examples. Run the storybook examples and navigate to a story like 'withKnobs - tweaks static values':
```
http://localhost:9010/?selectedKind=Addon%20Knobs.withKnobs&selectedStory=tweaks%20static%20values&downPanel=storybooks%2Fstorybook-addon-knobs
```

Delete the` 12.5` from the "Dollars" field. When viewing the console, the error described above will show, along with `A component is changing a controlled input of type number to be uncontrolled.`

### Please specify which version of Storybook and optionally any affected addons that you're running

`@storybook/react 3.2.17`
`@storybook/addon-knobs 3.2.17`

**Comment by Hypnosphi** [comment]
Should be quite easy to fix, feel free to send a PR

**Comment by M-ZubairAhmed** [comment]
hi can I work on this?

**Comment by Hypnosphi** [comment]
Of course @M-ZubairAhmed 
Make sure to refer to [`CONTRIBUTING.md`](https://github.com/storybooks/storybook/blob/master/CONTRIBUTING.md) to get familiar with our dev setup

**Comment by ah-adarlow** [comment]
A similar issue happens with date knobs as well.

**Comment by kaitmore** [comment]
Also experiencing this issue +1

**Comment by kaitmore** [comment]
hi @Hypnosphi! I have a PR for this, but I'm getting permission denied when I try to push to origin.

**Comment by CLL80** [comment]
@kaitmore, I don't think this is set up to allow direct PRs. You can fork the repo and set the PR to come from your fork, though.
